drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -8114,14 +8114,10 @@ void RenderingDeviceVulkan::compute_list_dispatch_indirect(ComputeListID p_list,
 }
 
 void RenderingDeviceVulkan::compute_list_add_barrier(ComputeListID p_list) {
-#ifdef FORCE_FULL_BARRIER
-	_full_barrier(true);
-#else
-	_memory_barrier(VK_PIPELINE_STAGE_COMPUTE_SHADER_BIT, VK_PIPELINE_STAGE_COMPUTE_SHADER_BIT, VK_ACCESS_SHADER_WRITE_BIT, VK_ACCESS_SHADER_READ_BIT, true);
```

Comment:
I'm a bit concerned that the ``dstAccessMask`` becomes ``VK_ACCESS_SHADER_READ_BIT | VK_ACCESS_SHADER_WRITE_BIT`` instead of just ``VK_ACCESS_SHADER_READ_BIT``. I'm not sure what the realistic performance impact would be though

Reply:
I think we can just assume that mid-list we don't expect write-write syncs, so I'm pushing with a change to honor that.